storage/range_command.go

```diff
@@ -118,77 +157,70 @@ func (r *Range) executeCmd(batch engine.Engine, ms *engine.MVCCStats, args proto
 	// the node to be classified as without further uncertain reads for the
 	// remainder of the transaction.
 	// See the comment on proto.Transaction.CertainNodes.
-	if tErr, ok := reply.Header().GoError().(*proto.ReadWithinUncertaintyIntervalError); ok && tErr != nil {
+	if tErr, ok := err.(*proto.ReadWithinUncertaintyIntervalError); ok {
 		// Note that we can use this node's clock (which may be different from
 		// other replicas') because this error attaches the existing timestamp
 		// to the node itself when retrying.
 		tErr.ExistingTimestamp.Forward(r.rm.Clock().Now())
 	}
 
-	// Return the error (if any) set in the reply.
-	return intents, err
+	return
 }
 
 // Get returns the value for a specified key.
-func (r *Range) Get(batch engine.Engine, args *proto.GetRequest, reply *proto.GetResponse) []proto.Intent {
-	val, intents, err := engine.MVCCGet(batch, args.Key, args.Timestamp, args.ReadConsistency == proto.CONSISTENT, args.Txn)
-	reply.Value = val
-	reply.SetGoError(err)
-	return intents
+func (r *Range) Get(batch engine.Engine, args *proto.GetRequest) (reply proto.GetResponse, intents []proto.Intent, err error) {
+	reply.Value, intents, err = engine.MVCCGet(batch, args.Key, args.Timestamp, args.ReadConsistency == proto.CONSISTENT, args.Txn)
+	return
```

Comment:
Heh, "As-is" was actually not using any returns at all. I went with named returns because the alternative was adding a `reply := proto.FooResponse{}` at the beginning of each method, and the `return` style was a smaller change to the existing `SetGoError()` style.